### ISSUE TITLE
Check consistency of optional parameter default value

### ISSUE THREAD

**Issue by thomaslevesque** [code-cracker/code-cracker#575]
When you implement an interface method with an optional parameter, specifying a different default value in the implementation has no effect when the method is called via the interface, so it's a potential bug. For instance:

```
interface IFoo { void Bar(string s = "hello"); }
class Foo : IFoo { public void Bar(string s = "hi") { Console.WriteLine(s); } }
IFoo foo = new Foo();
foo.Bar(); // will print "hello", not "hi"
```

I have written an analyzer for this (I know we're supposed to discuss it before actually doing it, but I wanted to check the feasibility before I mentioned it). I still have to write the code fix, but it shouldn't take long.

```
Severity : Warning
Category: Maintainability (I'm not really happy with that, but I can't find a more suitable category)
Diagnostic Id: CC0110
```

What do you think? Is this something we should include in Code Cracker?

**Comment by giggio** [comment]
@thomaslevesque I think it makes sense. I just verified the sample code. For me, it is a go ahead.
The category, feels like `Usage` for me. See examples of other analyzers [here](https://msdn.microsoft.com/en-us/library/ee1hzekz.aspx?f=255&MSPPError=-2147217396).

I have assigned the issue to you. Remember to fork and PR into the vnext branch.

**Comment by thomaslevesque** [comment]
I managed to write the code fix, but it was harder than I expected (mostly because there is no obvious way to format the `ExplicitDefaultValue` of the parameter).

I still have to write the tests, and I have a question: how do I test a code fix that has 2 possible actions? Is there an example somewhere in Code Cracker?

**Comment by thomaslevesque** [comment]
Ignore the question in my previous message, I just noticed the `codeFixIndex` parameter in `VerifyCSharpFixAsync` ;)

**Comment by giggio** [comment]
Closed on #578.
